plutus-ledger/src/Ledger/Orphans.hs

```diff
@@ -3,13 +3,44 @@
 {-# OPTIONS_GHC -Wno-orphans #-}
 module Ledger.Orphans where
 
+import           Codec.Serialise.Class   (Serialise, decode, encode)
+import           Crypto.Hash             (Digest, SHA256, digestFromByteString)
+import           Data.Aeson              (FromJSON (parseJSON), ToJSON (toJSON))
+import qualified Data.Aeson              as JSON
 import qualified Data.Aeson.Extras       as JSON
 import           Data.Bifunctor          (bimap)
+import qualified Data.ByteArray          as BA
+import qualified Data.ByteString         as BSS
 import qualified Data.Text               as Text
 import           Plutus.V1.Ledger.Bytes
 import           Plutus.V1.Ledger.Crypto
 import           Web.HttpApiData         (FromHttpApiData (..), ToHttpApiData (..))
 
+{- Note [Serialising Digests from Crypto.Hash]
+This is more complicated than you might expect.  If you say
+`encode = encode . BA.unpack` then the contents of the digest are
+unpacked into a `Word8` list with 32 entries.  However, when cborg
+serialises a list, every element in the output is preceded by a type
+tag (in this case, 24), and this means that the serialised version is
+about 64 bytes long, twice the length of the original data.  Packing
+the `Word8` list into a `ByteString` first fixes this because cborg
+just serialises it as a sequence of contiguous bytes. -}
+
+instance Serialise (Digest SHA256) where
+    encode = encode . BSS.pack . BA.unpack
+    decode = do
+      d :: BSS.ByteString <- decode
+      let bs :: BA.Bytes = BA.pack . BSS.unpack $ d
+      case digestFromByteString bs of
+        Nothing -> fail $ "Couldn't decode SHA256 Digest: " ++ show d
+        Just v  -> pure v
```

Comment:
@michaelpj  @j-mueller  This only thing that uses this function is an instance for the Render typeclass in `Wallet.Rollup.Render` from `plutus-contract`. And commenting that instance does not produce compile time error in plutus* folders (haven't checked in PS files). Seems like this could either  be removed, or migrated to cardano-crypto-class. What do you think?

Reply:
Fine, or move it to `Wallet.Rollup.Render` as an orphan, perhaps?